Let Barbarians produce cloth on maps without seafaring

Niektory opened the pull request:
Removed redundant check from the Weaving mill, so cloth can be produced on maps without seafaring but with waterways.
The building can't be built on maps without waterways and seafaring anyway so there's no need to check it during production.
Fixes #3699

Niektory committed:
Let Barbarians produce cloth on maps without seafaring

Removed redundant check from the Weaving mill.
The building can't be built on maps without waterways and seafaring anyway so there's no need to check it during production.

Noordfrees commented:
Code is straightforward, fix tested and working.

The `checkmap` command is then used only by shipyards, but those already have a `map_check` as well. Perhaps we should remove `checkmap` altogether?

Niektory commented:
It doesn't seem to be needed. Do you want me to nuke it?

Niektory committed:
Remove productionsite program "checkmap"

It is redundant because all buildings that use it also use map_check, which checks for the same thing at construction time.

Niektory commented:
[![Foo](https://user-images.githubusercontent.com/4470900/79287859-64061a00-7ec5-11ea-88d2-3dc9edd3ea79.png)](https://www.monkeyuser.com/2020/levels-of-satisfaction/)

Noordfrees reviewed (approved):
LGTM :)
Travis is happy except the usual GCC; transient appveyor failure. I'll merge now
